crates/iceberg/src/spec/values.rs

```diff
@@ -995,7 +995,7 @@ mod tests {
         assert_eq!(literal, expected_literal);
 
         let mut writer = apache_avro::Writer::new(&schema, Vec::new());
-        writer.append_ser(bytes).unwrap();
+        writer.append_ser(Into::<ByteBuf>::into(literal)).unwrap();
```

Comment:
There is a `ByteBuf::from(value: impl Into<Vec<u8>>)` which overrides the `From<Literal>` for `ByteBuf` and I need to somehow specify that I want the conversion from `Literal` to `ByteBuf`. Maybe you know of a different way to specify which `form/into` implementation to use.

Reply:
> There is a `ByteBuf::from(value: impl Into<Vec<u8>>)` which overrides the `From<Literal>` for `ByteBuf`

Hi, I'm not quite understanding. Does `Into<Vec<u8>>` exist for `Literal`? Or is it because `Literal` implements `From<Literal> for BytesBuf`?

Reply:
Oh, I got it. `ByteBuf` provides it's own `ByteBuf::from` that may conflicts with `From` trait. Maybe we should provide a `Literal::to_vec()` instead?

Reply:
> Oh, I got it. `ByteBuf` provides it's own `ByteBuf::from` that may conflicts with `From` trait.

Exactly.

It actually works with `writer.append_ser(ByteBuf::from(literal)).unwrap();` when I provide a `impl From<Literal> for Vec<u8>`.

Reply:
> The PR looks good to me overall, but I'm slightly concerned about the use of Into::<ByteBuf>::into(literal). Will our users have to write code in this manner? Or does it occur internally?

One choice is to impl `impl From<Literal> for Vec<u8>` instead of `impl From<Literal> for ByteBuf` and then we can convert to ByteBuf by `ByteBuf::from(literal)`

Another choice may be:
```
let bytes: BytesBuf = literal.into();
writer.append_ser(bytes)
```
Actually in the case where the type is BytesBuf explicitlly, we don't need ` Into::<ByteBuf>::into(literal)`.🤔

> when I provide a impl From<Literal> for Vec<u8>.

This way looks good to me.